docs/core/deploying/native-aot/index.md

```diff
@@ -134,6 +134,7 @@ Native AOT apps have the following limitations:
 - Implies compilation into a single file, which has known [incompatibilities](../single-file/overview.md#api-incompatibility).
 - Apps include required runtime libraries (just like [self-contained apps](../index.md#publish-self-contained), increasing their size as compared to framework-dependent apps).
 - <xref:System.Linq.Expressions> always use their interpreted form, which is slower than run-time generated compiled code.
+- Generic interface methods and generic virtual methods substituted with structs (e.g. `public abstract M<T>(T t)` vs. `public abstract M(object o)` or `public abstract M<T>(T t) where T : class`) are slower and consume more disk space than in CoreCLR. Widespread use may cause significant performance degredation.
```

Comment:
Do we have perf numbers saying they are slower? (I never really measured these myself.)

Generics over struct in general have to be expanded eagerly. I have seen big expansions in the past that didn't involve GVMs too.

I would probably put wording about eager generic expansion in the performance section of the docs, not as a limitation. It is a feature...

Cc @jkotas

Reply:
> Do we have perf numbers saying they are slower? (I never really measured these myself.)

GVM in NAOT implementation is not fundamentally different from GVM implementation in regular CoreCLR. There should not be a significant throughput degradation.

The throughput performance differences for GVMs should be on the same order of magnitude as for many other constructs. For example, access to static fields in types with constructors in NAOT can have performance differences of similar magnitude when compared to CoreCLR w/ tiered JIT.

Reply:
I think that’s fair, but the size implications are probably worth calling out. I’ve been telling people that they have to be careful about gvm size expansion, which is usually my red flag of, “if I’m explaining an architectural difference to people in issues, I should have documentation to point to it.”

Otherwise it seems like we’re not giving users the information they need to be successful.

Reply:
Tried to reword this to be more general. Left a comment specific to GVMs because people may not realize that each type gets its own instantiation -- two instantiations are not shared between the different types even if they have the same generic arguments.